magda-typescript-common/src/authorization-api/ApiClient.ts

```diff
@@ -2,56 +2,81 @@ require("isomorphic-fetch");
+        }
+    }
+
+    getMergeRequestInitOption(extraOptions: RequestInit = null): RequestInit{
+        let options: RequestInit = this.requestInitOption ? {...this.requestInitOption} : {};
+        if(!extraOptions) return options;
+        let headers = {...options.headers, ...extraOptions.headers};
+        options = {
+            ...options,
+            ...extraOptions
+        };
+        options.headers = headers;
+        return options;
+    }
+
+    async getUser(userId: string): Promise<Maybe<User>> {
+        return await this.handleGetResult(
+            fetch(`${this.baseUrl}/private/users/${userId}`, this.getMergeRequestInitOption())
         );
     }
 
-    getUserPublic(userId: string): Promise<Maybe<User>> {
-        return this.handleGetResult(
-            fetch(`${this.baseUrl}/public/users/${userId}`)
+    async getUserPublic(userId: string): Promise<Maybe<User>> {
+        return await this.handleGetResult(
+            fetch(`${this.baseUrl}/public/users/${userId}`, this.getMergeRequestInitOption())
         );
     }
 
-    lookupUser(source: string, sourceId: string): Promise<Maybe<User>> {
+    async lookupUser(source: string, sourceId: string): Promise<Maybe<User>> {
         return this.handleGetResult(
             fetch(
-                `${this
-                    .baseUrl}/private/users/lookup?source=${source}&sourceId=${sourceId}`
+                `${this.baseUrl}/private/users/lookup?source=${source}&sourceId=${sourceId}`,
+                this.getMergeRequestInitOption()
             )
         );
     }
 
-    createUser(user: User): Promise<User> {
-        return fetch(`${this.baseUrl}/private/users`, {
-            method: "POST",
-            headers: {
-                "Content-Type": "application/json"
-            },
-            body: JSON.stringify(user)
-        })
-            .then(res => res.json())
-            .then(res => {
-                if (res.status >= 400) {
-                    throw new Error(
-                        `Encountered error ${res.status} when POSTing new user to ${this
-                            .baseUrl}/private/users`
-                    );
-                }
-                return Object.assign({}, user, res);
-            })
-            .catch(e => {
-                console.error(e);
-                throw e;
-            });
+    async createUser(user: User): Promise<User> {
+        try{
+            const res = await fetch(`${this.baseUrl}/private/users`, this.getMergeRequestInitOption({
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                body: JSON.stringify(user)
+            }));
+            if(res.status >= 400) {
+                throw new Error(
+                    `Encountered error ${res.status} when POSTing new user to ${this.baseUrl}/private/users`
+                );
+            }
+            const resText = await res.text();
```

Comment:
you can use `res.json()` to automatically parse the json :)

Reply:
no problem. was for debugging purpose so that I can set a break point to check the actual body text. By the way, should we make AuthApi always return JSON response? At this moment, it return plain text upon error? If so, what should be the response structure? maybe a simple `{errorMessage:'xxxxxx'}`?

Reply:
done in commit a903ef188d2ce9e411d1df3794c8ec6cecb73f59
Will gradually modify relevant code in future to make sure Auth API always returns JSON response